protos/sigstore_rekor.proto

```diff
@@ -88,21 +89,21 @@ message InclusionPromise {
 // as described here https://www.rfc-editor.org/rfc/rfc9162#name-signed-certificate-timestam.
 message TransparencyLogEntry {
         // The index of the entry in the log.
-        int64 log_index = 1;
+        int64 log_index = 1 [(google.api.field_behavior) = REQUIRED];
         // The unique identifier of the log.
-        dev.sigstore.common.v1.LogId log_id = 2;
+        dev.sigstore.common.v1.LogId log_id = 2 [(google.api.field_behavior) = REQUIRED];
         // The kind (type) and version of the object associated with this
         // entry. These values are required to construct the entry during
         // verification.
-        KindVersion kind_version = 3;
+        KindVersion kind_version = 3 [(google.api.field_behavior) = REQUIRED];
         // The UNIX timestamp from the log when the entry was persisted.
-        int64 integrated_time = 4;
+        int64 integrated_time = 4 [(google.api.field_behavior) = REQUIRED];
         // The inclusion promise/signed entry timestamp from the log.
-        InclusionPromise inclusion_promise = 5;
+        InclusionPromise inclusion_promise = 5 [(google.api.field_behavior) = REQUIRED];
         // The inclusion proof can be used for online verification that the
         // entry was appended to the log, and that the log has not been
         // altered.
-        InclusionProof inclusion_proof = 6;
+        InclusionProof inclusion_proof = 6 [(google.api.field_behavior) = REQUIRED];
```

Comment:
@kommendorkapten , I'm inclined that the commend should read "The inclusion proof can be used for **offline** verification" rather than "online verification". WDYT?

Reply:
The inclusion proof requires that the client performs a request against the transparency log, so it's an online operation. The inclusion _promise_ (i.e verifying the SET) is what can be done offline.

Reply:
What is the purpose of having `InclusionProof inclusion_proof` then?
I thought the values in `InclusionProof inclusion_proof` enable `offline` verification.

Reply:
I would make both optional. At least one should be required for this message, but there's no way to easily represent this.

Currently Rekor returns both an inclusion proof and promise always, but I think this behavior shouldn't be reflected in the specs. Future iterations of the log may choose to return either a promise or an inclusion proof or both, but I don't think we should guarantee that a promise is always set. 

Something else to consider is that the log index may not be set for an inclusion promise, because the entry has yet to be admitted into the log. Now, once again the current behavior of Rekor is to always put an entry into the log immediately, but this may not always be the case. I'd rather have clients handle it as optional for now, it'll be less burden in the future if we do make it optional. Thoughts?

Reply:
Does that mean that if Rekor omits the proof, then offline verification would not be possible?

By the way, is there something like 'caching Rekor proxy' like there are proxy servers for artifact repositories?

Reply:
So what's tricky here is there is what Rekor does currently, and what I want Rekor to do in the future. So my first question is, should the protos reflect the current state of the service, or the ideal state?

Slight aside, right now a promise is simply a signed statement from Rekor. Even with a set of hashes to calculate an inclusion proof, there is no guarantee that an entry actually exists in Rekor without doing an online lookup. This is because you need to verify consistency (that the log is still append-only), which would require an online lookup. I've been thinking about how to improve this, but we're awhile away from being able to have stronger offline verification.

With what we've got right now, a client could "verify" offline using either the hashes from the proof or the promise, they're pretty much equivalent. Currently Rekor will return both because of its design, but technically it could omit one or the other - A promise would be omitted when a proof is requested (because if a proof can be generated, why need a promise?), or a proof would be omitted when only a promise can be generated if the entry is actively being uploaded (Rekor operates in a batched mode, but the batch size = 1 currently, so entries are immediately uploaded)

So I don't really know the best way to handle this for whether it should be required or optional. I'd lean towards being permissive and allow either the proof or promise or both, but not require that a promise is always present.

For proxies, we've chatted a little bit about Rekor read-only replicas. I see this as a requirement for being able to scale out Rekor. This is definitely something I want to do, we just need to scope it out and figure out technical blockers.

Reply:
@haydentherapper I think this is good as-is now, as it reflects the current way Rekor works? 

As there are more changes expected in Rekor (SET v2), omitting the log index I think we can wait with them until Rekor is updated?